<a href="https://colab.research.google.com/github/goodladcici/mom/blob/main/mfin7037_hw1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# MFIN7037  Quantitative Trading 
- JIAYUN HUANG 3035877531

- TIANCI XIE 3035942269

- YIJIE PAN 3035914341




## Part 1: The market model
---



### a.
Firstly, it is related to behavioral finance, where investors either overreact to important information or underreact to it (Subrahmanyam, 2018). Investors tend to react properly regarding to the news that confirms their beliefs, high return will continue having good performence, while underract to news that differs from their belief (Antoniou et al., 2013). As a result, stocks with good performence have strong resistance for bad news. In addition, it is suggested many investors are relying on overly simplified models, with which they make persistant forecast errors and thus lead to momentum (Hong et al., 2007). Secondly, it is a risk premium story, where investors gain return as the compenstation for market risk that they are taking, proportional to beta. 

### b. 

Assuming it is a monthly strategy, theoretically, 
>Annualized average return  $=r_{monthly}\times{12} = {\frac{∑^{n}_{i=1}r_i}{n}}\times{12}$, where n is the number of months in dataset.

>Annualized volatility  $=\sigma_{monthly} \times \sqrt{12} = {\sqrt {\frac {\sum^{n}_{i=1}(r_{i}-{\bar{r}})^{2}}{n}}}\times \sqrt{12}$, where $\sigma$ implies standard deviation of returns

>Annualized sharpe ratio $=Sharpe\_Ratio_{monthly} \times \sqrt{12} = (r_{monthly}/\sigma_{monthly})\times \sqrt{12}$

The annualized average return, volatility and sharpe ratio are  0.06964, 0.1519 and 0.4584 respactively.


In [ ]:
import pandas as pd # data frames wrangling
import numpy as np # fast and vectorized math functions
import statsmodels.formula.api as smf
import os # operating system operations like check files existance

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

In [ ]:
# Set the directory
os.chdir('/content/drive/MyDrive/data')

In [ ]:
# read mkt_rf from ff_four_factor_monthly.parquet
ff4 = pd.read_parquet('ff_four_factor_monthly.parquet')
ff4.rename(columns = {'month_end':'date'},inplace = True)

In [ ]:
def sharpe(x):
    return x.mean() / x.std()

sharpe = sharpe(ff4['mom'])
return_mean = ff4['mom'].mean()
return_volatility = ff4['mom'].std()

print('annualized sharpe ratio : {}'.format(sharpe * 12**0.5))
print('annualized mean return : {}'.format(return_mean*12))
print('annualized return volatility : {}'.format(return_volatility * 12**0.5))

annualized sharpe ratio : 0.458373184773055
annualized mean return : 0.06964130684901665
annualized return volatility : 0.15193145926172957


### c.
There are several reasons for the discrepancy. First, the macroeconomic factors, which vary by geography and trading window, affect the momentum return. Second, the impact of trading cost and tax are not considered during the paper's calculation of strategy return. Since the strategy is based on information 11 months, when something happened between the information collection time and trade time, the startgy is unable to cater to the change.

The trading costs for AQR is likely to be lower, for the reason that they might operate as market maker, from which they can acquire trading rebate such as lower transaction cost. In addition, with more comprehensive infrastructure compared with retail traders, they might pay lower implicit trading cost such as bid-ask spread. 

### d.
For value weight strategy, the momentum goes up by -0.2542 ($\beta_1$ ), i.e. goes down by 0.2542.

### e.
For the regression code, please refer to the code above.

The $\beta_{1}$ is -0.2542.

In [ ]:
# value weight regression 
smf.ols('mom ~ 1 + mkt_rf',data=ff4).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    mom   R-squared:                       0.096
Model:                            OLS   Adj. R-squared:                  0.095
Method:                 Least Squares   F-statistic:                     121.0
Date:                Tue, 12 Apr 2022   Prob (F-statistic):           7.96e-27
Time:                        07:02:25   Log-Likelihood:                 2017.0
No. Observations:                1147   AIC:                            -4030.
Df Residuals:                    1145   BIC:                            -4020.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0076      0.001      6.079      0.000       0.005       0.010
mkt_rf        -0.2542      0.023    -11.000      0.000      -0.300      -0.209
==============================================================================
Omnibus:                      219.484   Durbin-Watson:                   2.004
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             1341.229
Skew:                          -0.731   Prob(JB):                    5.70e-292
Kurtosis:                       8.092   Cond. No.                         18.8
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

### f.
Alpha denotes the excess return that an active portfolio manager achieves above the benchmarks. It is calculated as the percent by which the investment beat expectation. 

The annualized extra return is 9.12% ($12\times α$).


### g.
t value for $\beta_1$ to market is -11   , t value for $\alpha$ is 6.079, which are statistically significant.


### h.
No. The expectation of the momentum strategy return will go up by $\beta_1 × (-4\%) $

## Part 2: The Fama-French (1993) three-factor mode
---

### a.

No, the change of momentum will not be expected to $\beta_{1}$. Because benchmark risk factors are not independent, SMB and HML will change as market goes up. Assuming that stocks across the market goes up by 1%, the corresponding rising of SMB and HML are $x_1$ and $x_2$. Then the momentum will be expected to go up by $0.01×\beta_{1}+ x_{1}\times\beta_{2}+x_2\times\beta_{3}$. 

### b.

SMB stands for "Small (market capitalization) Minus Big" and HML for "High (book-to-market ratio) Minus Low"; they measure the historic excess returns of small caps over big caps and of value stocks over growth stocks.

SMB earns positive return by longing publicly traded companies with small market caps, whose return is usually higher than those with large market cap. HML accounts for value stocks with high book-to-market ratios that generate higher returns in comparison to the market.

### c.

The benchmarking against SMB and HML would be appropriate. 

As we known about Fama-French three factor model, size could be interpreted as a liquidity premium,and value multiples are compensation for default risk. Therefore the values of SMB and HML can reflects the risk of default and liqudity. This could explain the rationale for momentum trading strategies in $mkt_{Europe}, mkt_{HongKong}$ and  $mkt_{USA}$. 

In investment opportunity set theory. the value of an enterprise is the value generated by the ongoing future surplus of its existing operating assets, plus the value generated by all future investment plans that may expect a positive net present value, which is the so-called investment opportunity set. 

SMB explains the impact of company size on company value, while HML explains the impact of investment opportunity set on company value.

## Part 3: Being an Alpha Detective
---

### 1.

The exposures of this strategy to Fama French factors:

> $r_{mkt}-r_{f}$ : -0.5

> $smb$ : -0.5

> $hml$ : 0

> $rmw$ : -0.5

> $cma$ : 0.5


In [ ]:
# import dataframe h1_mystery_meat.parquet
df = pd.read_parquet('h1_mystery_meat.parquet')

In [ ]:
# apply regression on fama-french factors and mystery strategy
smf.ols('mystery_meat ~ 1 + mkt_rf + smb + hml + rmw + cma', data=df).fit().summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:           mystery_meat   R-squared:                       1.000
Model:                            OLS   Adj. R-squared:                  1.000
Method:                 Least Squares   F-statistic:                 1.295e+10
Date:                Mon, 11 Apr 2022   Prob (F-statistic):               0.00
Time:                        09:57:22   Log-Likelihood:                 7835.9
No. Observations:                 703   AIC:                        -1.566e+04
Df Residuals:                     697   BIC:                        -1.563e+04
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.0010   1.39e-07  -7217.444      0.000      -0.001      -0.001
mkt_rf        -0.5000    3.4e-06  -1.47e+05      0.000      -0.500      -0.500
smb           -0.5000   4.81e-06  -1.04e+05      0.000      -0.500      -0.500
hml         4.686e-06   6.29e-06      0.745      0.457   -7.67e-06     1.7e-05
rmw           -0.5000   6.53e-06  -7.66e+04      0.000      -0.500      -0.500
cma            0.5000   9.68e-06   5.16e+04      0.000       0.500       0.500
==============================================================================
Omnibus:                      114.064   Durbin-Watson:                   1.996
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               27.508
Skew:                           0.099   Prob(JB):                     1.06e-06
Kurtosis:                       2.052   Cond. No.                         82.5
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

### 2.

According to the summary of the regression, it can be concluded that the mystery stategy is conducted by four portfolios, $r_{mkt}-r_{f}, smb, rmw, cma$, and transaction fee,which is the intercept of the model. The reason is that $\beta_{hml}$ is too small to consider and its t-statistic is not significant, while each $\beta$ of other factors has significant t-statisctic, which implies the evidence of the relationship with the mystery strategy is very strong. 

The model should be:


>$r_{mystery} = -0.5 \times (r_{mkt}-r_{f}) - 0.5 \times smb - 0.5 × rmw + 0.5 × cma - 0.001$ 

In summary, the mystery meat is a strategy shorting $1/2$ units market portfolio ($r_{mkt}$), $1/2$ units smb portfolio ($r_{smb}$) and $1/2$ units rmw portfolio ($r_{rmw}$), and longing $1/2$ units cma portfolio ($r_{cma}$) and $1/2$ units risk free portfolio ($r_{f}$) ,with $0.1\%$ monthly transaction fee. Shortly, if someone holds 1-dollar mystery meat portfolio, it equivalents to shorting 0.5 dollars marktet porfolio, smb portfolio and rmw portfolio, then buying 0.5 dollars cma portfolio, with $0.1\%$ monthly transaction fee.

##*Reference:*
---

*Avanidhar Subrahmanyam (2018). Equity market momentum: A synthesis of the literature and suggestions for future work. Pacific-Basin Finance Journal, Volume 51, 291-296.*

*Antoniou, C., Doukas, J., & Subrahmanyam, A. (2013). Cognitive Dissonance, Sentiment, and Momentum. Journal of Financial and Quantitative Analysis, 48(1), 245-275.*

*Hong, H., Stein, J., & Yu, J. (2007). Simple Forecasts and Paradigm Shifts. The Journal of Finance (New York), 62(3), 1207-1242.*

*Vafiadis, N. (2015). Forecasting volatility and the risk-return tradeoff: An application on the fama-french benchmark market return. Journal of Time Series Econometrics, 7(2), 181-216. doi:http://dx.doi.org/10.1515/jtse-2012-0018*

*Leippold, Markus, and Roger Rueegg. "Fama–French Factor Timing: The Long‐only Integrated Approach." European Financial Management : The Journal of the European Financial Management Association 27.4 (2021): 666-700. Web.*